project/src/main/java/com/google/sps/servlets/SellInvestmentServlet.java

```diff
@@ -0,0 +1,124 @@
+package com.google.sps.servlets;
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+/**
+ * Receive data when an investment is sold and updated the investments and participants database.
+ */
+@WebServlet("/sellInvestment")
+public class SellInvestmentServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(SellInvestmentServlet.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long investmentID = -1, priceSold = -1;
+    Date today = new Date();
+    long userID = -1;
+    long compID = -1;
```

Comment:
Setting these variables to -1 and using them later to check the SQL query return seems like a really counterintuitive way of validating the existence of the investment in database. Can you do this some other way? Also as best I can see you only use userID = -1, why set the other variables to anything?

Reply:
![image](https://user-images.githubusercontent.com/75593542/107587021-e8503780-6c65-11eb-8770-088f46ce6f28.png)
I can't not just leave the variables not initialised that's why I set all of it to -1.

Reply:
Yeah because the code logic is a bit confusing, so it's not clear the compiler that that situation will trigger the early return. What if you used if (rs.next()) instead of the while loop? Then the else will catch the situation in which there is no investment and you can not worry about initialising weird numbers to your variables. I just think it would make things easier to understand.

Reply:
Yup! changed the position of the variables and removed the for loop. 2aaa2e98d521ea45b07c0423eafd386899d42c80